src/App.tsx

```diff
@@ -1,31 +1,28 @@
 import React, { useState } from "react";
 import "./App.css";
-import { ImportModal } from "./Components/ImportModal/ImportModal";
-import { Header } from "./Components/Shared/Header";
-import { Container } from "react-bootstrap";
-import { VideoModal } from "./Components/VideoModal/VideoModal";
 import type { MediaPresenter } from "./Components/MediaPresenter";
-
+import {Modal} from "./Components/Modal/Modal";
+import {Homepage} from "./Components/Homepage/Homepage";
 type AppProps = {
   mediaPresenter: MediaPresenter;
 };
+
 const App: React.FC<AppProps> = ({mediaPresenter}) => { 
-  const [show, setShow] = useState<boolean>(false);
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const closeModal = () => {
+    setModalOpen(false);
+  }
+
+  const openModal = ()  => {
+    setModalOpen(true);
+    if (mediaPresenter.mediaReady()) {
```

Comment:
what is this `if` statement for?

Reply:
there should have been a function called inside but it's implemented in a presenter that's not pushed yet. My bad should have put a "//TODO" instead

Replying to #2, I added the modal section because I thought that's part of the homepage, would you like me to remove it?